### PR TITLE
Improve test_common coverage

### PR DESCRIPTION
**Pull request by akaszynski**
### Improve `test_common` coverage

This PR introduces several tests to improve the code coverage of `test_common.py`.  Major items include:

- Not testing the the code coverage of methods inherited by subclasses of `Common`
- Fixing a bug where field data is not updated.
- Cleans up some error types (i.e. getting rid of generic `Exception`)
- Fixes `test_transform`.  Before test was raising an error due to improper test setup.
- Tests setting the extent for uniform grids.
- Adds testing for edge cases (i.e. setting cell data with a range or non-contiguous arrays).
- Some minor spelling fixes.
- Not including code coverage of depreciated functions with `# pragma: no cover`.  We should either test them and expect a warning, or ignore the code coverage.

This PR is the start of an effort to improve code coverage to the point we no longer need doc-testing to get full code coverage.

### CODE REVIEW

**Comment by akaszynski** [comment]
@GuillaumeFavelier 
It should be noted that the qt-plotting module should really be moved out of `pyvista`.  I'd really like a `pyvista-qt` module to be created, as unit testing for this is driving me insane, especially when local testing involves waiting for dialog boxes to popup and close.  Also, I noticed that `pyqt` would hang unless there was some mouse movement in some cases (which might be causing unattended testing to fail under MacOS).

Considering the number of times I have to restart testing on different OSes and the increasing complexity of the code, I hope there's some support for moving qt out of `pyvista`.  With `pyvista-qt` we can specify minimum versions, and even have a `pyvista-pyside` module as well, all while keeping the `BasePlotter` class intact.

File: pyvista/core/common.py [hunk 1]
```diff
@@ -584,7 +587,7 @@ def _activate_texture(mesh, name):
             keys = list(mesh.textures.keys())
             # Grab the first name available if True
             idx = 0 if not isinstance(name, int) or name is True else name
-            if idx > len(keys):
+            if idx > len(keys):  # is this necessary?
```
Comment (banesullivan) [inline pyvista/core/common.py:590]: This should probably be taken out.

**Comment by banesullivan** [comment]
As for your other comment above, YES! Let's move all qt stuff into a new downstream package. Isn't this what #614 is about?

The Qt stuff is just becoming too much to handle here... we need to separate it out and abstract the `BackgroundPlotter` interface so it could be implemented with PyQt or PySide2. 

See also: #628, https://github.com/pyvista/pyvista/issues/562#issuecomment-623039858, and all the other issues here that detail problems about PyQt

**Comment by JevinJ** [comment]
@akaszynski ipython should be added to the testing requirements, test_print_repr(True, True) failed for me until I manually added it.

**Comment by akaszynski** [comment]
> @akaszynski ipython should be added to the testing requirements, test_print_repr(True, True) failed for me until I manually added it to my venv.

Interesting that it's failing for you as it's running on our test servers.  Did you run:
```
pip install -r requirements_test.txt
```

While `ipython` isn't a direct dependency, looking at it I saw:
`itkwidgets` --> `ipywidgets` --> `ipython`

**Comment by JevinJ** [comment]
> Interesting that it's failing for you as it's running on our test servers. Did you run:

Hmm, maybe PyCharm doesn't pick it up because it's not "requirements.txt" specifically, I can try a fresh install and doing that myself to verify it will work.
